go/vt/tabletserver/query_executor.go

```diff
@@ -106,8 +106,46 @@ func (qre *QueryExecutor) Execute() (reply *mproto.QueryResult) {
 			conn := qre.getConn(qre.qe.connPool)
 			defer conn.Recycle()
 			reply = qre.execSQL(conn, qre.query, true)
+		}
+		if reply != nil {
```

Comment:
goto would work, for sure. But before doing it, I make another attempt: define a execDmlAutoCommit func and call it in the default clause for the non transaction case. I feel that might give us a better code flow than one uses goto.